lib/go/thrift/http_transport.go

```diff
@@ -40,14 +41,23 @@ func NewThriftHandlerFunc(processor TProcessor,
 
 // gz transparently compresses the HTTP response if the client supports it.
 func gz(handler http.HandlerFunc) http.HandlerFunc {
+	sp := &sync.Pool{
+		New: func() interface{} {
+			return gzip.NewWriter(nil)
+		}}
+
 	return func(w http.ResponseWriter, r *http.Request) {
 		if !strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") {
 			handler(w, r)
 			return
 		}
 		w.Header().Set("Content-Encoding", "gzip")
-		gz := gzip.NewWriter(w)
-		defer gz.Close()
+		gz := sp.Get().(*gzip.Writer)
+		gz.Reset(w)
+		defer func() {
+			_ = gz.Close()
```

Comment:
also nit: For writes `Close` usually also includes `Flush` so ignore the error could be bad. But we were ignoring the error before this change, so this at least won't make things worse. We don't have a way to return error here anyways.

Reply:
Yeah I thought about it but couldn't think of a clean way of handling the error. I'm open to suggestions :)

Reply:
Reading code I think the best we could do is probably pass in a logger through `NewThriftHandlerFunc` and log the error if it's non-nil, but that's not very useful and it's also a breaking change so probably not worth doing.